constants.gradle

```diff
@@ -16,10 +16,10 @@ project.ext {
     releaseVersion = '2.16.1'
     releaseVersionCode = 2016001
     minSdkVersion = 16
-    appTargetSdkVersion = 29
+    appTargetSdkVersion = 31
     // Upgrading this requires [Internal ref: b/193254928] to be fixed, or some
     // additional robolectric config.
-    targetSdkVersion = 30
+    targetSdkVersion = 31
```

Comment:
Looking at the internal referenced issue, it appears this will cause tests to fail. Updating to Robolectric 4.7.0 or later will resolve some of the issues (and is now possible). Looking at the notes in the issue, it looks like two problems will remain after doing that:

1. Taking the Robolectric update causes heap space exhaustion when running the tests. This is because adding an additional SDK version to Robolectric increases baseline memory usage, because it needs to load a whole new set of Android classes during runtime. This is relatively easily fixed with some changes to the `build.gradle` files to request larger-than-default heap space when running the tests.
1. ExoPlayer defaults to using `MediaCodec` in asynchronous mode on API 31 or later, but the Robolectric `MediaCodec` shadow doesn't implement this mode yet. We'll need to add some logic to make sure tests that fail as a result are not run on API 31+. Alternatively, we can configure the player to always use synchronous `MediaCodec` mode specifically for the tests, or (least preferred) wait for Robolectric support.

So there's a bit of work to be done here. Feel free to take a look on your side. We'll keep this pull request open in the meantime.